### PR TITLE
Feat/layout opinion related articles

### PR DESCRIPTION
**Pull request by colincclark**
https://nidigitalsolutions.jira.com/browse/REPLAT-1615

- adds images to opinion template for related articles (there needs to be a future story to fix these images on the storybook for native as they currently do not display properly) - https://github.com/newsuk/times-components/issues/791)
- styles opinion related articles
- adds opinion byline to article summary for new opinion byline UI
- extend DatePublication component to show or hide the day within the date (a requirement for all related articles)
- fixes numerous layout issues across devices (tested on Firefox, Safari, Chrome, IE and native)
- changes the fixed article width at tablet and desktop breakpoints as directed by Jake Cox

## Chrome
opinion 2

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/2f316914b1c8e8152851bfb630c877eee2091913/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/0251e68455ec2a43881ea346befe0b08226e447e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/8fe438031a1095544e8f6395a135b9aacd383e94/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7b8695016a8793498989312f65c04abb24ad938b/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7456bb6f0dfb3ba0de554db8f190dcf63d924683/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/2c62763b50639072a13a13ea9292d90954c7a3a0/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/0402d100336300700bdf6af92d8040cda5599857/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/89419d4edf69f7ce438a36dbbf1eadd56ba3815a/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/d4e2d81c29d36f0b20ba220a16f585b90f58b65a/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/82ed2d4a5ab01a5324c83a341f2e4edf52dd7739/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/5c58f78c3e48ecdd9d18b45d245513ae2b13105c/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f1e16692dc47b9b71e527b8af36e53f90e962cd3/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/5d858acef0ac1875acbcb5bb77c7809dbaf1b464/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/27128522ad223d0330e10580ba8d61299f8baf53/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/a4b09c91ac95fd2bb375d97c2516240a8cadefc6/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3700c3eb11f5ac477ab6facf51b874d8e16ca696/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/e9d2614eb97436cfaed8cc624abf605aeb46cb4c/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/9492e24f45f85e49b862f2527e60f2b3ab77b49e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/690c6a36b516cfba1cb092e5cc77e9042b8c50c2/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/4a6cf42416055b675efa109f79987a8380a145b6/index.html

File: packages/article/related-articles/related-articles.js [hunk 1]
```diff
@@ -63,6 +75,15 @@ const RelatedArticles = ({ articles, mainId, onPress, template }) => {
             }
           />
         );
+      case "OPINION_AND_TWO":
+        return (
+          <OpinionAndTwoSlice
+            opinion={(config = {}) => renderArticleItem(config, mainArticle)}
```
Comment (aronshamash) [inline packages/article/related-articles/related-articles.js]: should this be `main` now?

File: packages/card/styles/index.js [hunk 2]
```diff
@@ -5,16 +5,31 @@ const styles = StyleSheet.create({
...
+  },
+  reversedCardContainer: {
+    height: "auto"
   },
   imageContainer: {
     flex: 1,
     minWidth: "100%"
   },
+  reversedImageContainer: {
+    marginBottom: -10,
```
Comment (aronshamash) [inline packages/card/styles/index.js]: `spacing(-2)`

File: packages/responsive-styles/responsive-styles.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import styled from "styled-components";
 
 export { keyframes } from "styled-components";
 export const config = {
-  mediumBpWidth: "84%",
-  wideBpWidth: "58%"
+  mediumBpWidth: "80.8%",
```
Comment (aronshamash) [inline packages/responsive-styles/responsive-styles.js]: this is global change can you describe the reason?

File: packages/slice/templates/leadandtwo/config.web.js [hunk 4]
```diff
@@ -69,19 +78,17 @@ export const getConfigWrapper = ({ supportCount }) => {
         }
 
         .leadContentContainerClass {
-          flex-grow: 1;
           flex-basis: 0 !important;
+          flex-grow: 1;
+          margin-bottom: 10px;
```
Comment (aronshamash) [inline packages/slice/templates/leadandtwo/config.web.js]: spacing(1)

File: packages/card/__tests__/shared.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+import { shallow } from "enzyme";
+import Card from "../../card";
+import Loading from "../card-loading";
+
+const cardProps = {
+  image: {
+    uri:
+      "https://www.thetimes.co.uk/imageserver/image/methode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F9242e576-4dfc-11e7-a20e-a11097d3353d.jpg?crop=1463%2C975%2C293%2C12"
+  },
+  imageRatio: 1.5,
```
Comment (aronshamash) [inline packages/card/__tests__/shared.js]: `2 / 3`

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/dc01ecebc1cf4d612e37d992984ec8e59a6bb337/index.html

File: packages/article/related-articles/related-articles.js (hunk 1 above)
Comment (colincclark) [reply packages/article/related-articles/related-articles.js]: nope, this is the renderProp

File: packages/card/styles/index.js (hunk 2 above)
Comment (colincclark) [reply packages/card/styles/index.js]: groan....I forgot about this

File: packages/responsive-styles/responsive-styles.js (hunk 3 above)
Comment (colincclark) [reply packages/responsive-styles/responsive-styles.js]: Jake asked me to

File: packages/slice/templates/leadandtwo/config.web.js (hunk 4 above)
Comment (colincclark) [reply packages/slice/templates/leadandtwo/config.web.js]: yes

File: packages/card/__tests__/shared.js (hunk 5 above)
Comment (colincclark) [reply packages/card/__tests__/shared.js]: why? this is a test..

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/e42639b0e6bf46f4812ddde015eef93d8de7d204/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/01298af34df5c3f0d4315c0c975365dd61e7b07f/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/53b84444b7a26359cfbf9436c425475ca2718cd9/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/01848babcc2dfed4792ce10c28af587e81d58930/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/72f0339b9ccd3dc289d2caae0623a2a9c0830030/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/94c01e29965efdf07903642e3ba283d90fce21ce/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/04eb1727862ed1c792ee2fc757553f604e499153/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/b86a5fe249e0e7ee0e0d8e2181606a0fa83900c1/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/00eecf1e26fda21ceeed3860b37ef9550b1618ff/index.html

File: packages/slice/templates/opinionandtwo/config.web.js [hunk 6]
```diff
@@ -20,89 +27,175 @@ export const getOpinionConfig = ({ itemCount }) => ({
...
-    `,
-    mediumUp: () => {
-      const withSupportsImageStyle = `
-        margin-bottom: 10px;
-        min-width: 270px;
+    base: () => {
+      const getOpinionImageStyle = () => {
+        if (supportCount === 0)
+          return `
+          margin-bottom: -10px;
```
Comment (aronshamash) [inline packages/slice/templates/opinionandtwo/config.web.js]: `margin-bottom: spacing(-2);`

**Pull request merged**